pandas/core/groupby/generic.py

```diff
@@ -1643,8 +1643,12 @@ def _wrap_aggregated_output(
         DataFrame
         """
         indexed_output = {key.position: val for key, val in output.items()}
-        name = self._obj_with_exclusions._get_axis(1 - self.axis).name
-        columns = Index([key.label for key in output], name=name)
+
+        agg_axis = self._obj_with_exclusions._get_axis(1 - self.axis)
+        if isinstance(agg_axis, MultiIndex):
+            columns = Index([key.label for key in output], names=agg_axis.names)
```

Comment:
does this line alone work? I thought we fixed construction for this. alt does `ensure_index` work here? I *really* do not like special casing like this *any*where except in the index module itself (and even then its not great).

Reply:
It does! I've removed the if-else clause